src/components/App/App.js

```diff
@@ -24,11 +25,44 @@ type Props = {
   selectedProjectId: ?Project,
 };
 
-class App extends PureComponent<Props> {
+type State = {
+  isNodeJsMissing: boolean,
+};
+
+const { dialog, shell } = remote;
+
+class App extends PureComponent<Props, State> {
+  state = {
+    isNodeJsMissing: false,
+  };
+
   componentDidMount() {
-    window.addEventListener('beforeunload', this.killAllRunningProcesses);
+    const logAppLoaded = node_version => {
+      logger.logEvent('load-application', {
+        node_version: node_version || 'missing',
+      });
+    };
+
+    checkIfNodeIsAvailable()
+      .then(result => {
+        logAppLoaded(result.trim());
+        this.setState({ isNodeJsMissing: false });
+      })
+      .catch(error => {
+        this.setState({
+          isNodeJsMissing: true,
+        });
+        dialog.showErrorBox(
+          'Node missing',
+          'It looks like Node.js isn\'t installed. Node is required to use Guppy.\nWhen you click "OK", you\'ll be directed to instructions to download and install Node.'
+        );
+        shell.openExternal(
+          'https://github.com/joshwcomeau/guppy/blob/master/README.md#installation'
+        );
+        logAppLoaded();
+      });
```

Comment:
One last thing, totally optional :) I saw that some people use async/await inside their lifecycle methods. I think we could try doing something like this instead:

```js
async componentDidMount() {
  try {
    const nodeVersion = await checkIfNodeIsAvailable();

    logger.logEvent(...);
    this.setState({ wasSuccessfullyInitialized: true });
  } catch (e) {
    ... error handling
  }
}
```

This isn't something we've done in our codebase. I'm up for trying it, but it's totally your call :)

Reply:
Ah, sure, I just wasn't sure if we're using async/await. I'll change it.